src/Microsoft.VisualStudio.AppDesigner/PropPageDesigner/PropPageDesignerDocData.vb

```diff
@@ -215,6 +220,9 @@ Namespace Microsoft.VisualStudio.Editors.PropPageDesigner
         Public Function LoadDocData2(pszMkDocument As String) As Integer Implements IVsPersistDocData2.LoadDocData
             'Nothing to do here, no real file to load
             _mkDocument = pszMkDocument
+            ' Sign up for RDT events so we know when to dispose ourselves - there is no point doing this until we have the document moniker to compare to
```

Comment:
Dispose should be done in response to IVsPersistDocData.Close, that's how folks around the tree do it.

Reply:
Thank you! An excellent suggestion and change. I didn't think to look at that because the current Dispose calls Close() on the wrapped text buffer via `IVsPersistDocData` so I thought the relationship went the other way, but you're right the RDT calls Close anyway so there is no need for events!